File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.js
/**
 * @param {string} colors
 * @param {number[]} neededTime
 * @return {number}
 */
var minCost = function(colors, neededTime) {
    let count = 0;

    for(let i=1; i<colors.length;i++){
        while(i<colors.length && colors.charAt(i)==colors.charAt(i-1)){
            count += Math.min(neededTime[i],neededTime[i-1]);
            if(neededTime[i]<neededTime[i-1])
                neededTime[i] = neededTime[i-1];
         
            i++;
        }
    }

    return count;
};

File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int count = 0;

        for(int i=1; i<neededTime.length;i++)
            while(i<colors.length() && colors.charAt(i)==colors.charAt(i-1)){
                
                if(neededTime[i]<neededTime[i-1]){
                    count += neededTime[i];
                    neededTime[i] = neededTime[i-1];
                }
                else
                    count += neededTime[i-1];

                i++;
            }

        return count;
    }
}